bin/prover-client/src/ckp_runner.rs

```diff
@@ -0,0 +1,143 @@
+use std::sync::Arc;
+
+use jsonrpsee::{
+    core::{client::ClientT, params::ArrayParams},
+    http_client::HttpClient,
+    rpc_params,
+};
+use strata_rpc_types::HexBytes;
+use tokio::time::{interval, Duration};
+use tracing::{error, info, warn};
+use uuid::Uuid;
+
+use crate::{
+    dispatcher::TaskDispatcher,
+    primitives::tasks_scheduler::ProvingTaskStatus,
+    proving_ops::checkpoint_ops::{CheckpointOperations, CheckpointOpsParam},
+    task::TaskTracker,
+};
+
+/// Continuously checks for the latest checkpoint index.
+/// Dispatches tasks when a new checkpoint is detected and discards the current proving tasks.
+pub async fn run_checkpoint_runner(
+    sequencer_client: HttpClient,
+    ckp_task_dispatcher: TaskDispatcher<CheckpointOperations>,
+    task_tracker: Arc<TaskTracker>,
+) {
+    info!("Checkpoint runner started");
+    let mut current_checkpoint_idx = 0;
+    let mut is_started = false;
+    let mut ticker = interval(Duration::from_secs(1));
+    let mut current_task_id = Uuid::default();
+
+    loop {
+        ticker.tick().await;
+
+        let new_checkpoint_idx =
+            fetch_latest_checkpoint_index(&sequencer_client, current_checkpoint_idx).await;
+
+        if new_checkpoint_idx > current_checkpoint_idx || !is_started {
+            match handle_new_checkpoint(new_checkpoint_idx, &ckp_task_dispatcher, &task_tracker)
+                .await
+            {
+                Ok(task_id) => {
+                    current_task_id = task_id;
+                    current_checkpoint_idx = new_checkpoint_idx;
+                    is_started = true;
+                }
+                Err(_) => {
+                    warn!(
+                        "Failed to handle new checkpoint idx {:?}",
+                        new_checkpoint_idx
+                    );
+                }
+            }
+        }
```

Comment:
This can be made simpler with following:
```
to_fetch_idx = 0

loop {
    latest = get_latest_checkpoint()
    if latest >= to_fetch_idx {
        handle_checkpoint(latest)?
        to_fetch_idx = latest + 1
    }
}```